src/main/java/de/dafuqs/spectrum/blocks/fluid/DragonrotFluidBlock.java

```diff
@@ -39,43 +40,20 @@ public void randomDisplayTick(BlockState state, World world, BlockPos pos, Rando
 	public boolean canPathfindThrough(BlockState state, BlockView world, BlockPos pos, NavigationType type) {
 		return false;
 	}
-	
-	/**
-	 * @param world The world
-	 * @param pos   The position in the world
-	 * @param state BlockState of the mud. Included the height/fluid level
-	 * @return Dunno, actually. I just mod things.
-	 */
-	public boolean receiveNeighborFluids(World world, BlockPos pos, BlockState state) {
-		for (Direction direction : Direction.values()) {
-			BlockPos blockPos = pos.offset(direction);
-			if (world.getFluidState(blockPos).isIn(FluidTags.WATER)) {
-				world.setBlockState(pos, SpectrumBlocks.SLUSH.getDefaultState());
-				this.playExtinguishSound(world, pos);
-				return false;
-			} else if (world.getFluidState(blockPos).isIn(FluidTags.LAVA)) {
-				world.setBlockState(pos, Blocks.BLACKSTONE.getDefaultState());
-				this.playExtinguishSound(world, pos);
-				return false;
-			} else if (world.getFluidState(blockPos).isIn(SpectrumFluidTags.MUD)) {
-				world.setBlockState(pos, Blocks.MUD.getDefaultState());
-				this.playExtinguishSound(world, pos);
-				return false;
-			} else if (world.getFluidState(blockPos).isIn(SpectrumFluidTags.LIQUID_CRYSTAL)) {
-				world.setBlockState(pos, SpectrumBlocks.ROTTEN_GROUND.getDefaultState());
-				this.playExtinguishSound(world, pos);
-				return false;
-			} else if (world.getFluidState(blockPos).isIn(SpectrumFluidTags.MIDNIGHT_SOLUTION)) {
-				world.setBlockState(pos, SpectrumBlocks.BLACK_SLUDGE.getDefaultState());
-				this.playExtinguishSound(world, pos);
-				return false;
-			}
+
+	public BlockState handleFluidCollision(World world, FluidState state, FluidState otherState) {
```

Comment:
Methods that may return null should be annotated as @Nullable.

Looking fine, otherwise